### PR TITLE
Resolve #132

### LINKED ISSUE #132: peer-dependency of ember-cli 3.2
Did you all mean to add a peer dependency of `"ember-cli": "~3.2.0"`? https://github.com/quaertym/ember-cli-dependency-checker/compare/v3.2.0...v3.3.0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R57

That seems overly specific and is quite old (July of 2018 https://github.com/ember-cli/ember-cli/releases/tag/v3.2.0).
It also is causing new errors in our build chain akin to:
```console
buildDepsnpm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: zysales-web@0.0.4
npm ERR! Found: ember-cli@4.3.0
npm ERR! node_modules/ember-cli
npm ERR!   dev ember-cli@"^4.3.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer ember-cli@"~3.2.0" from ember-cli-dependency-checker@3.3.0
npm ERR! node_modules/ember-cli-dependency-checker
npm ERR!   dev ember-cli-dependency-checker@"^3.2.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/circleci/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/circleci/.npm/_logs/2022-04-08T20_28_14_327Z-debug-0.log
```

Any help is appreciated.

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
https://github.com/quaertym/ember-cli-dependency-checker/issues/132
https://stackoverflow.com/questions/71811139/not-able-to-create-a-new-ember-app-could-not-resolve-dependency-error?noredirect=1#comment126901983_71811139

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
     "sinon": "^13.0.0"
   },
   "peerDependencies": {
-    "ember-cli": "~3.2.0 || ^4.0.0"
```
Comment (SergeAstapov) [inline package.json:57]: @NullVoxPopuli shouldn't this be `^3.2.0 || ^4.0.0`? I think there is no reason to prevent Ember CLI v3 users

Comment (bertdeblock) [reply package.json:57]: The `peerDependencies` entry was added a long time ago, but was never released until v3.3.0. 
Maybe the safest thing to do, is to completely remove it for now and then release a patch? 
We can always re-add it later and release a new major just to be sure?

**Comment by bertdeblock** [comment]
@quaertym, mind having a look at this PR? 
The current peer dependency statement breaks the `ember new` flow for NPM users. Specifically, the `~3.2.0` part.

**Comment by quaertym** [comment]
Let's go with `^3.2.0 || ^4.0.0` as suggested above.

**Comment by NullVoxPopuli** [comment]
Is there a test somewhere to ensure this problem doesn't happen again?

**Comment by quaertym** [comment]
Not sure what's the best way to test this. But I think it is safe until ember-cli 5.x is released.

**Comment by NullVoxPopuli** [comment]
I think, this would be a sufficient test to add in your CI:
```
ember new my-app --skip-npm
cd my-app
# remove line with your dependency
sed  '/"ember-cli-dependency-checker"/d' ./package.json
npm install
npm add --save-dev ember-cli-dependency-checker@file:../foo/bar/wherever
npm install
```
as long as everything has exit status code 0, that should be sufficient.

**Comment by quaertym** [comment]
@rwjblue any input on how to test this?

**Comment by quaertym** [comment]
BTW, I released this as `v3.3.1`.

**Comment by SergeAstapov** [comment]
@quaertym I assume we can have ember-try scenarios with different versions of `ember-cli`, including `latest`/`beta`?

**Pull request merged**